shell/browser/api/atom_api_top_level_window.cc

```diff
@@ -118,6 +121,8 @@ TopLevelWindow::TopLevelWindow(v8::Isolate* isolate,
 }
 
 TopLevelWindow::~TopLevelWindow() {
+  all_windows.erase(std::remove(all_windows.begin(), all_windows.end(), this),
+                    all_windows.end());
```

Comment:
Since you left this unresolved, I'll expand on the previous comment with rationale:

 * If using a set or unordered set, this line would be a very simple `all_windows_.erase(this);`

 * It looks like `all_windows_` doesn't care about the order of the items and the same window will never occur twice, so `unordered_set` seems like the simplest & fastest container.

That said, the current impl should run. If you have a reason for usign a vector, e.g. some future work you intend to do that works better with a vector, :+1:

Reply:
cc @MarshallOfSound, do you want to address this prior to merge?